### PR TITLE
Support `PQinitOpenSSL()`.

### PR DESCRIPTION
**Pull request by jtv**
Been meaning to do this for ages, though there hasn't been much demand for it.  It came up again in #675.

### CODE REVIEW

**Comment by jtv** [comment]
@KayEss what do you think — does this API make sense?  I've created an enum for things you _don't_ want initialised, and you can call `pqxx::do_not_initialize_ssl()` passing zero or more of these enum values.

File: include/pqxx/connection.hxx [hunk 1]
```diff
@@ -91,6 +91,14 @@ concept ZKey_ZValues = std::ranges::input_range<T> and requires(T t)
...
 == 2;
 #endif // PQXX_HAVE_CONCEPTS
+
+
+/// Control OpenSSL/crypto library initialisation.
+/** @param flags a bitmask of `1 << flag` for each of the `skip_init` flags.
+ *
+ * Ignores the `skip_init::noop` flag.
+ */
+void PQXX_COLD PQXX_LIBEXPORT do_not_initialize_ssl(int flags) noexcept;
```
[the remark this answers is not in the document]
Comment (jtv) [reply include/pqxx/connection.hxx:101]: Thanks for checking!

The idea here is to make it work like varargs but without the problem of how to terminate the args list.  What I hate about using bitwise "or" on an enum is that I see no good way to relate the parameter type to the enum type.

So I tried something new and yes, it's clunky.  Now I'm tempted to try it as compile time arguments, as a parameter pack.

[the remark this answers is not in the document]
Comment (KayEss) [reply include/pqxx/connection.hxx:101]: What would the use for the current design look like?

 If you want to have something that looks like varargs then you could use `std::initializer_list<skip_init>`, and if you use an `enum class` then the `skip_` part also takes care of itself: `skip_init::none` or `skip_init::ssl` etc

[the remark this answers is not in the document]
Comment (jtv) [reply include/pqxx/connection.hxx:104]: @KayEss I finally got back to this and rewrote it to take the skip flags as template args.  Which lets us have nice, straightforward typing on the arguments.

File: include/pqxx/connection.hxx [hunk 2]
```diff
@@ -111,6 +119,60 @@ class const_connection_largeobject;
...
+/** When a running process makes its first SSL connection to a database through
+ * libpqxx, libpq automatically initialises the OpenSSL and libcrypto
+ * libraries.  But there are scenarios in which you may want to suppress that.
+ *
+ * This enum is a way to express this.
+ */
+enum skip_init : int
+{
+  /// A do-nothing flag that does not affect anything.
+  noop,
```
[the remark this answers is not in the document]
Comment (jtv) [reply include/pqxx/connection.hxx:136]: I've renamed it to `skip_init_ssl()`.

File: include/pqxx/connection.hxx [hunk 3]
```diff
@@ -111,6 +119,60 @@ class const_connection_largeobject;
...
+
+  /// Skip initialisation of OpenSSL library.
+  openssl,
```
[the remark this answers is not in the document]
Comment (jtv) [reply include/pqxx/connection.hxx:139]: Well it's an enum class, so you pretty much can't refer to the values without also mentioning the name of the enum.

File: include/pqxx/connection.hxx (hunk 2 above)
[the remark this answers is not in the document]
Comment (jtv) [reply include/pqxx/connection.hxx:136]: @KayEss what do you think?  I've added a test and the call site looks kind of sensible to me.

[the remark this answers is not in the document]
Comment (jtv) [reply include/pqxx/connection.hxx]: Thanks @KayEss for your help in getting this right!

**Pull request merged**